tests/playwright/src/model/pages/settings-bar.ts

```diff
@@ -44,9 +44,11 @@ export class SettingsBar {
   }
 
   public async openTabPage<T extends SettingsPage>(type: new (page: Page) => T): Promise<T> {
-    const desiredPage = new type(this.page);
-    await (await desiredPage.getTab()).click();
-    return desiredPage;
+    return await test.step(`Open ${type.name} tab from Settings`, async () => {
```

Comment:
wondering on the pattern, if it's ok to return the promise, not the await of the promise (as the signature of the method is to return a promise)

Reply:
I'm not really sure what you mean. The await there is for test.step, test.step is an async call which needs to be awaited.

Reply:
I'm not speaking about playwright, I'm speaking about TypeScript

if you have a method returning a promise, you return a promise, not the await of the promise.
as anyway the caller of the method will await on it.

```ts
async myMethod: Promise<unknown> {
  return step(); // that returns the promise
}
```


or, if you just need to wait, then the signature of the method is Promise<void>

and then you can just do:

```ts
async myMethod: Promise<void> {
  await step();
}
```

Reply:
ok I understand what you mean, but this is the current practice we've used in the test framework and I'd rather not have two different approaches at the same time without some sort of understanding.

We can certainly refactor it to the way you suggest, however I don't think that it's in scope for this PR.

Reply:
> We can certainly refactor it to the way you suggest, however I don't think that it's in scope for this PR.

but the code before this PR was not using any 'return await', it's part of this PR that is introducing the `return await` so it's in the scope of this PR

I'm asking how playwright behave if you just return the promise without awaiting it

using
```
return test.step(`Open ${type.name} tab from Settings`, async () => {
```

instead of
```
return await test.step(`Open ${type.name} tab from Settings`, async () => {
```

Reply:
Both those forms will work because in the end the call is awaited in the actual test, there won't be a functional difference in the end.

Reply:
so why are we not using less code ?

Reply:
Because as I mentioned this was the approach that was taken in the test framework, the same argument can be made for most(all?) async calls in there so I didn't want to implement a new approach just because it would be slightly less code. As I said we can certainly refactor it to this way, functionally there won't be a difference but I think that requires consensus on it, not just changing it suddenly in an unrelated PR.